esp32-hal-uart.c:153:9: warning: 'return' with no value, in function returning non-void

### Board

ESP32-S2

### Device Description

ESP32-S2 Mini

### Hardware Configuration

No

### Version

v2.0.13

### IDE Name

Arduino IDE 2.2.1

### Operating System

Windows 10

### Flash frequency

N/A

### PSRAM enabled

yes

### Upload speed

N/A

### Description

Compile warning on 2.0.13 due to missing return value. Seem related to #8629  

### Sketch

```cpp
N/A
```


### Debug Message

```plain
_C:\Users\Stefan\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.13\cores\esp32\esp32-hal-uart.c: In function 'uartSetPins':
C:\Users\Stefan\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.13\cores\esp32\esp32-hal-uart.c:153:9: warning: 'return' with no value, in function returning non-void
         return;
         ^~~~~~
C:\Users\Stefan\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.13\cores\esp32\esp32-hal-uart.c:149:6: note: declared here
 bool uartSetPins(uint8_t uart_num, int8_t rxPin, int8_t txPin, int8_t ctsPin, int8_t rtsPin)_
```


### Other Steps to Reproduce

_No response_

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

--- a/cores/esp32/esp32-hal-uart.c
+++ b/cores/esp32/esp32-hal-uart.c
@@ -150,7 +150,7 @@ bool uartSetPins(uint8_t uart_num, int8_t rxPin, int8_t txPin, int8_t ctsPin, in
 {
     if(uart_num >= SOC_UART_NUM) {
         log_e("Serial number is invalid, please use numers from 0 to %u", SOC_UART_NUM - 1);
-        return;
+        return false;
     }
 
     // IDF uart_set_pin() will issue necessary Error Message and take care of all GPIO Number validation.

